server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesResponseTests.java

```diff
@@ -118,6 +123,35 @@ public void testFailureSerialization() throws IOException {
         }
     }
 
+    /**
+     * check that failure serialization between different minor versions after 7.13 works
+     */
+    public void testMixedVersionFailureSerialization() throws IOException {
+        FieldCapabilitiesResponse original = createResponseWithFailures();
+        FieldCapabilitiesResponse deserialized;
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            Version outVersion = VersionUtils.randomVersionBetween(random(), Version.V_7_13_0, Version.CURRENT);
+            output.setVersion(outVersion);
+            original.writeTo(output);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), getNamedWriteableRegistry())) {
+                Version inVersion = VersionUtils.randomVersionBetween(random(), Version.V_7_13_0, Version.CURRENT);
+                in.setVersion(inVersion);
+                deserialized = new FieldCapabilitiesResponse(in);
+                assertEquals(-1, in.read());
+            }
+        }
+        assertThat(deserialized.getIndices(), Matchers.equalTo(original.getIndices()));
+        // only match size of failure list and indices, most exceptions don't support 'equals'
+        List<FieldCapabilitiesFailure> deserializedFailures = deserialized.getFailures();
+        assertEquals(deserializedFailures.size(), original.getFailures().size());
+        int i = 0;
+        for (FieldCapabilitiesFailure originalFailure : original.getFailures()) {
+            FieldCapabilitiesFailure deserializedFaliure = deserializedFailures.get(i);
+            assertThat(deserializedFaliure.getIndices(), Matchers.equalTo(originalFailure.getIndices()));
+            i++;
+        }
+    }
```

Comment:
having written some similar tests before, what I usually do is also write a test that reads a true format from the previous version that we are testing, otherwise it is really fake, you are only testing that the same version can read from that same version when the stream is set to the previous version. I generally have the base64 string in the test itself and make sure that it can be deserialized.